_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -24,34 +25,37 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any [HTML element][] with a [semantic][semantic role] [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any non-empty `th` element that is [included in the accessibility tree][], has a [semantic role][] of [rowheader][] or [columnheader][], and for which all of the following are true:
 
-- the element is [visible][]; and
-- the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][] is [included in the accessibility tree][].
+- **in a table** the `th` element is a [descendant][] of a `table` element for which all of the following are true:
+  - **visible** the `table` element is [visible][]; and
+  - **minimum rows** the `table` element's [`rows` collection](https://html.spec.whatwg.org/multipage/tables.html#dom-table-rows) has 2 or more `tr` elements; and
```

Comment:
🤔 Not fully sure about that one.
```html
<table>
<tr>
  <th>Opening hours</th> <td>8-17</td>
</tr>
</table>
```
sounds like a reasonable case.
And
```html
<table>
<tr>
  <th>Opening hours</th> <th>Mon-Fri</th> <td>8-17</td>
</tr>
</table>
```
sounds like a bad one (where the first `th` is not assigned to anything due to the second one (or maybe it is still assigned to the second `th`, can't remember…))

(OTOH, restricting Applicability won't create false positives, so it won't really hurt…)

Reply:
Hey @Jym77 ,
Your second example is actually a common thing where the data is yet to be added. As long as visually it is clear that there are no other rows in this table I don't think that would be a failure of 1.3.1. I think that we are going to catch more issues with this rule in the actual two dimensional tables.

Happy to work on it though if you think this is the direction we should follow.

Reply:
Agree with Jean-Yves. The first case is reasonable, and I can't come up with a good reason for us to ignore it.

Reply:
The question is how much value we get out of that example? Not too much if you think that there is only one row and row headers are useful when you navigate between the rows. This change corresponds to Wilco's [survey](https://docs.google.com/spreadsheets/d/1OSkPFocXk4K3zYLnwS78WLsWO4PvE5yRcsauyefuIUI/edit#gid=1850701681) comments. I'm trying to avoid the below scenario:
 ``` 
      <table>
        <thead>
          <tr>
          <td>N/a
          <th>2008
          <th>2007
          <th>2006
        <tbody>
          <tr>
          <td>
          <td>
          <td>
          <td>
      </table>
```

Because tables are two dimensional the most we will get out of this rule if there will be at least two rows and there will be a non-empty data cell in a row that follows a column header row or in any of the rows providing that there are row headers in them.

Would you agree?

Reply:
I don't understand how your example applies to this condition (your example has 2 rows, therefore this conditions applies) :-)

But I can live with this condition. As Jean-Yves mentioned, it will only limit what is applicable, not create false positives. Although I still can't see why we should not consider the one row example Jean-Yves described.

Reply:
The problem that the above example introduces is that there is no use of the `<th>` elements, technically they do not have an assigned `<td>` element and they should not be column headers but the implementation varies. My condition requires at least one non-empty `<td>` element in the second row, meaning that it is clear visually that there is a relationship between a data cell and a column header cell.